code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,197 @@
+
+/datum/select_equipment/proc/make_outfit_entries(category="General", list/L)
+	var/list/entries = list()
+	for(var/path in L)
+		var/datum/outfit/O = path
+		var/priority = 0
+		if(path == /datum/outfit/job/roboticist) //dummy check here until I get favorites working
+			priority = 1
+		entries += list(OUTFIT_ENTRY(category, path, initial(O.name), priority))
+	return entries
+
+
+//GLOB.custom_outfits lists outfit *objects* so we'll need to do some custom handling for it
+/datum/select_equipment/proc/make_custom_outfit_entries(list/L)
+	var/list/entries = list()
+	for(var/datum/outfit/O in L)
+		entries += list(OUTFIT_CUSTOM_ENTRY("Custom", REF(O), O.name, 0)) //it's either this or special handling on the UI side
+	return entries
+
+/datum/select_equipment/ui_data(mob/user)
+	var/list/data = list()
+	if(!dummy)
+		init_dummy()
+
+	var/datum/preferences/prefs
+	if(target.client)
+		prefs = target.client.prefs
+	var/icon/dummysprite = get_flat_human_icon(null, prefs=prefs, dummy_key = dummy_key, outfit_override = selected_outfit)
+	data["icon64"] = icon2base64(dummysprite)
+	data["name"] = target
+
+	var/list/custom
+	custom += make_custom_outfit_entries(GLOB.custom_outfits)
+	data["custom_outfits"] = custom
+	data["current_outfit"] = selected_name
+	return data
+
+
+/datum/select_equipment/ui_static_data(mob/user)
+	var/list/data = list()
+	if(!cached_outfits)
+		cached_outfits = list()
+		cached_outfits += list(OUTFIT_ENTRY("General", /datum/outfit, "Naked", 2))
+		cached_outfits += make_outfit_entries("General", subtypesof(/datum/outfit) - typesof(/datum/outfit/job) - typesof(/datum/outfit/plasmaman))
+		cached_outfits += make_outfit_entries("Jobs", typesof(/datum/outfit/job))
+		cached_outfits += make_outfit_entries("Plasmamen Outfits", typesof(/datum/outfit/plasmaman))
+
+	data["outfits"] = cached_outfits
+	return data
+
+
+/datum/select_equipment/proc/resolve_outfit(text)
+
+	var/path = text2path(text)
+	if(ispath(path, /datum/outfit))
+		return path
+
+	else //don't bail yet - could be a custom outfit
+		var/datum/outfit/custom_outfit = locate(text)
+		if(istype(custom_outfit))
+			return custom_outfit
+
+
+/datum/select_equipment/ui_act(action, params)
+	if(..())
+		return
+	. = TRUE
+	switch(action)
+		if("preview")
+			var/datum/outfit/O = resolve_outfit(params["path"])
+
+			if(ispath(O)) //got a typepath - that means we're dealing with a normal outfit
+				selected_name = O //these are keyed by type
+				//by the way, no, they can't be keyed by name because many of them have duplicate names
+
+			else if(istype(O)) //got an initialized object - means it's a custom outfit
+				selected_name = REF(O) //and the outfit will be keyed by its ref (cause its type will always be /datum/outfit)
+
+			else //we got nothing and should bail
+				return
+
+			selected_outfit = O
+
+		if("applyoutfit")
+			var/datum/outfit/O = resolve_outfit(params["path"])
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names